File: backend/app/models/feed.py
from pydantic import BaseModel


class FeedClip(BaseModel):
    clip_id: str
    artist_id: str
    artist_name: str
    artist_avatar_url: str | None = None
    track_title: str
    video_url: str
    duration: str
    likes: int = 0
    created_at: str


class FeedResponse(BaseModel):
    clips: list[FeedClip]
    total: int


class InfluenceRequest(BaseModel):
    artist_id: str
    clip_id: str
    tag: str


class TagCount(BaseModel):
    tag: str
    count: int


class ArtistInfluencesResponse(BaseModel):
    artist_id: str
    total_influences: int
    tags: list[TagCount]

File: frontend/src/types/feed.ts
export interface FeedItem {
  clip_id: string
  artist_id: string
  artist_name: string
  artist_avatar_url: string | null
  track_title: string
  video_url: string
  duration: string
  likes: number
  created_at: string
}

export interface InfluenceTag {
  tag: string
  is_trending?: boolean
}

export interface InfluenceState {
  selected_tag: string | null
  is_sending: boolean
  confirmation_shown: boolean
}
